core/server/worker/src/test/java/alluxio/worker/page/PagedBlockWriterTest.java

```diff
@@ -0,0 +1,151 @@
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.page;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import alluxio.client.file.cache.CacheManager;
+import alluxio.client.file.cache.DefaultPageMetaStore;
+import alluxio.client.file.cache.LocalCacheManager;
+import alluxio.client.file.cache.PageId;
+import alluxio.client.file.cache.PageMetaStore;
+import alluxio.client.file.cache.PageStore;
+import alluxio.client.file.cache.evictor.CacheEvictor;
+import alluxio.client.file.cache.evictor.FIFOCacheEvictor;
+import alluxio.client.file.cache.store.LocalPageStoreDir;
+import alluxio.client.file.cache.store.LocalPageStoreOptions;
+import alluxio.client.file.cache.store.PageStoreOptions;
+import alluxio.conf.Configuration;
+import alluxio.conf.InstancedConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.util.CommonUtils;
+import alluxio.util.WaitForOptions;
+import alluxio.util.io.BufferUtils;
+
+import com.google.common.collect.ImmutableList;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+@RunWith(Parameterized.class)
+public class PagedBlockWriterTest {
+  private static final long BLOCK_ID = 1L;
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][] {
+        /*file_length, chuck_size, page_size*/
+        {2048, 1024, 128},
+        {2049, 1024, 128},
+        {2048, 1023, 128},
+        {2048, 1024, 129},
+    });
+  }
+
+  @Parameterized.Parameter
+  public int mFileLength;
+
+  @Parameterized.Parameter(1)
+  public int mChunkSize;
+
+  @Parameterized.Parameter(2)
+  public int mPageSize;
+
+  private LocalCacheManager mCacheManager;
+  private InstancedConfiguration mConf = Configuration.copyGlobal();
+  private PageMetaStore mPageMetaStore;
+  private CacheEvictor mEvictor;
+  private LocalPageStoreOptions mPageStoreOptions;
+  private PageStore mPageStore;
+  private LocalPageStoreDir mPageStoreDir;
+  private PagedBlockWriter mWriter;
+
+  @Rule
+  public ExpectedException mThrown = ExpectedException.none();
+
+  @Before
+  public void before() throws Exception {
+    mConf.set(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE, mPageSize);
+    mPageStoreOptions = (LocalPageStoreOptions) PageStoreOptions.create(mConf).get(0);
+    mPageStore = PageStore.create(mPageStoreOptions);
+    mEvictor = new FIFOCacheEvictor(mConf);
+    mPageStoreDir = new LocalPageStoreDir(mPageStoreOptions, mPageStore, mEvictor);
+    mPageStoreDir.reset();
```

Comment:
We can use a temp folder to ensure the cache dir is always empty before the test begins, and the test files are auto cleaned up after the test.